modules/yieldmoBidAdapter.js

```diff
@@ -476,51 +480,69 @@ function validateVideoParams(bid) {
 
   const isDefined = val => typeof val !== 'undefined';
   const validate = (fieldPath, validateCb, errorCb, errorCbParam) => {
-    const value = utils.deepAccess(bid, fieldPath);
-    if (!validateCb(value)) {
-      errorCb(fieldPath, value, errorCbParam);
+    if (fieldPath.includes('video')) {
+      const valueFieldPath = 'params.' + fieldPath;
+      const mediaFieldPath = 'mediaTypes.' + fieldPath;
+      const valueParams = utils.deepAccess(bid, valueFieldPath);
+      const mediaTypesParams = utils.deepAccess(bid, mediaFieldPath);
+      const hasValidValueParams = validateCb(valueParams);
+      const hasValidMediaTypesParams = validateCb(mediaTypesParams);
+
+      if (valueParams && hasValidValueParams) return valueParams;
+      else if (mediaTypesParams && hasValidMediaTypesParams) return hasValidMediaTypesParams;
+      else {
+        if (!hasValidValueParams) errorCb(valueFieldPath, valueParams, errorCbParam);
+        if (!hasValidMediaTypesParams) errorCb(mediaFieldPath, mediaTypesParams, errorCbParam);
+      }
+    } else {
+      const value = utils.deepAccess(bid, fieldPath);
+      if (!validateCb(value)) {
+        errorCb(fieldPath, value, errorCbParam);
+      }
+      return value;
     }
-    return value;
   }
 
   try {
+    validate('video.context', val => !utils.isEmpty(val), paramRequired);
+
     validate('params.placementId', val => !utils.isEmpty(val), paramRequired);
 
-    validate('mediaTypes.video.playerSize', val => utils.isArrayOfNums(val, 2) ||
+    validate('video.playerSize', val => utils.isArrayOfNums(val, 2) ||
       (utils.isArray(val) && val.every(v => utils.isArrayOfNums(v, 2))),
     paramInvalid, 'array of 2 integers, ex: [640,480] or [[640,480]]');
 
-    validate('mediaTypes.video.mimes', val => isDefined(val), paramRequired);
-    validate('mediaTypes.video.mimes', val => utils.isArray(val) && val.every(v => utils.isStr(v)), paramInvalid,
+    validate('video.mimes', val => isDefined(val), paramRequired);
+    validate('video.mimes', val => utils.isArray(val) && val.every(v => utils.isStr(v)), paramInvalid,
       'array of strings, ex: ["video/mp4"]');
 
-    validate('params.video', val => !utils.isEmpty(val), paramRequired);
+    validate('video', val => !utils.isEmpty(val), paramRequired);
```

Comment:
@ym-abaranov we don't need this check here, because you already have `validate('video.context',...`